crates/zizmor/src/audit/anonymous_definition.rs

```diff
@@ -0,0 +1,68 @@
+pub(crate) struct AnonymousDefinition;
+
+use crate::{
+    finding::{Confidence, Persona, Severity, location::Locatable as _},
+    state::AuditState,
+};
+
+use super::{Audit, AuditLoadError, Job, audit_meta};
+
+// Workflows without a name can be hard to find in the GitHub UI, so
+// severity is set higher than for Job.
+const ANONYMOUS_DEFINITION_WORKFLOW_SEVERITY: Severity = Severity::Low;
+const ANONYMOUS_DEFINITION_JOB_SEVERITY: Severity = Severity::Informational;
+
+audit_meta!(
+    AnonymousDefinition,
+    "anonymous-definition",
+    "workflow or action definitions without a name"
+);
+
+impl Audit for AnonymousDefinition {
+    fn new(_state: &AuditState<'_>) -> Result<Self, AuditLoadError> {
+        Ok(Self)
+    }
+
+    fn audit_workflow<'doc>(
+        &self,
+        workflow: &'doc crate::models::workflow::Workflow,
+    ) -> anyhow::Result<Vec<crate::finding::Finding<'doc>>> {
+        let mut findings = vec![];
+
+        if workflow.name.is_none() {
+            findings.push(
+                Self::finding()
+                    .severity(ANONYMOUS_DEFINITION_WORKFLOW_SEVERITY)
+                    .confidence(Confidence::High)
+                    .persona(Persona::default())
```

Comment:
Nit: Low confidence, but I _think_ we should consider making this pedantic-only to start -- the security impact is hard to substantiate, so having it be pedantic-only is a good way to let it bake for a bit (and moving to the regular persona is always easy + not considered a breaking change 🙂)

Reply:
Done in https://github.com/zizmorcore/zizmor/pull/937/commits/ef3d8f3546466954308e3faf682b5aff86016d66

Reply:
Went ahead and also removed the separate snapshot test, since we use pedantic for both now https://github.com/zizmorcore/zizmor/pull/937/commits/ecfdcfdadf42e874b2a96618c97f41384ee23069